src/components/Timer.tsx

```diff
@@ -19,10 +19,12 @@ const Timer = ({ duration, navigation, screenHidden }: ITimerProps) => {
   let playbackInstance: null | Audio.Sound = null;
 
   //Incrementer
-  const incrementTimer = () => {
+  const incrementTimer = async () => {
     if (count === 0) {
+      const streak = await updateStreak();
       navigation.navigate("Completion", {
-        duration: duration,
+        duration,
```

Comment:
Hmm, what if the play button is pressed though and we finish early? I don't think we would hit this branch..

Reply:
Correct, put a call in the finishEarly method